gensim/models/tfidfmodel.py

```diff
@@ -390,7 +390,7 @@ def initialize(self, corpus):
         self.num_nnz = numnnz
         self.dfs = dfs
         # and finally compute the idf weights
-        n_features = max(dfs) if dfs else 0
+        n_features = len(dfs) if dfs else 0
```

Comment:
dfs is never None, so we can use len without the ternary operator.

```suggestion
        n_features = len(dfs)
```

Furthermore, I think n_features is a useless variable here. We're defining it here, only to use it once in the logging statement below.

Reply:
I agree.